apis/python/test/test_distance_metrics.py

```diff
@@ -0,0 +1,120 @@
+import os
+
+import numpy as np
+import pytest
+from array_paths import *
+from sklearn.neighbors import NearestNeighbors
+
+from tiledb.vector_search import _tiledbvspy as vspy
+from tiledb.vector_search.ingestion import ingest
+from tiledb.vector_search.utils import load_fvecs
+
+siftsmall_uri = siftsmall_inputs_file
+queries_uri = siftsmall_query_file
+
+
+def normalize_vectors(vectors):
+    norms = np.linalg.norm(vectors, axis=1, keepdims=True)
+    return vectors / norms
+
+
+def test_cosine_similarity(tmp_path):
+    index_uri = os.path.join(tmp_path, "sift10k_flat_FLAT")
+    index = ingest(
+        index_type="FLAT",
+        index_uri=index_uri,
+        source_uri=siftsmall_uri,
+        source_type="FVEC",
+        distance_metric=vspy.DistanceMetric.COSINE,
+    )
+
+    dataset_vectors = load_fvecs(siftsmall_uri)
+    query_vectors = load_fvecs(queries_uri)
+
+    nn_cosine_sklearn = NearestNeighbors(n_neighbors=5, metric="cosine")
+    nn_cosine_sklearn.fit(dataset_vectors)
+    distances_sklearn, indices_sklearn = nn_cosine_sklearn.kneighbors(query_vectors)
+
+    indices_yourlib = index.query(query_vectors, k=5)
+
+    assert np.allclose(
+        distances_sklearn, indices_yourlib[0], 0.01
+    ), "Cosine similarity distances do not match"
```

Comment:
Can we lower `0.01`? Seems pretty big to me - if it's just about float precision, I've generally been able to use `1e-6` and have tests pass. But even a bit lower would be nice. Same below.